typescript/web/src/components/export-button/export-modal/__tests__/index.tsx

```diff
@@ -1,104 +1,41 @@
-
-import { incrementMockedDate } from "@labelflow/dev-utils/mockdate";
-import { processImage } from "../../../../connectors/repository/image-processing";
 import { mockNextRouter } from "../../../../utils/router-mocks";
+import { BASIC_DATASET_DATA } from "../../../../utils/tests/data.fixtures";
+
+initMockedDate();
 
 mockNextRouter({
-  query: { datasetSlug: "test-dataset", workspaceSlug: "local" },
+  query: {
+    datasetSlug: BASIC_DATASET_DATA.slug,
+    workspaceSlug: BASIC_DATASET_DATA.workspace.slug,
+  },
 });
 
 import { ExportModal } from "..";
+import { getApolloMockLink } from "../../../../utils/tests/apollo-mock";
 import { theme } from "../../../../theme";
-import { client } from "../../../../connectors/apollo-client/schema-client";
-import { setupTestsWithLocalDatabase } from "../../../../utils/setup-local-db-tests";
-import {
-  CREATE_TEST_DATASET_MUTATION,
-  CREATE_TEST_IMAGE_MUTATION,
-} from "../../../../utils/tests/mutations";
-import { CREATE_LABEL_MUTATION } from "../../../../connectors/undo-store/effects/shared-queries";
-import { LabelCreateInput } from "../../../../graphql-types/globalTypes";
-
-setupTestsWithLocalDatabase();
-
-jest.mock("../../../../connectors/repository/image-processing");
-const mockedProcessImage = processImage as jest.Mock;
+import { APOLLO_MOCKS } from "../export-modal.fixtures";
 
 const wrapper = ({ children }: PropsWithChildren<{}>) => (
-  <ApolloProvider client={client}>
+  <MockedProvider link={getApolloMockLink(APOLLO_MOCKS)}>
     <ChakraProvider theme={theme} resetCSS>
       {children}
     </ChakraProvider>
-  </ApolloProvider>
+  </MockedProvider>
 );
 
-const createDataset = async (
-  datasetId = "mocked-dataset-id",
-  name = "test dataset"
-) => {
-  return await client.mutate({
-    mutation: CREATE_TEST_DATASET_MUTATION,
-    variables: { datasetId, name, workspaceSlug: "local" },
-  });
-};
-
-const labelData = {
-  geometry: {
-    type: "Polygon",
-    coordinates: [
-      [
-        [0, 0],
-        [1, 0],
-        [1, 1],
-        [0, 1],
-        [0, 0],
-      ],
-    ],
-  },
-};
-
-const createLabel = (data: LabelCreateInput) => {
-  return client.mutate({
-    mutation: CREATE_LABEL_MUTATION,
-    variables: {
-      data,
-    },
-  });
-};
-
-const imageWidth = 500;
-const imageHeight = 900;
-
-const createImage = async (name: String) => {
-  const mutationResult = await client.mutate({
-    mutation: CREATE_TEST_IMAGE_MUTATION,
-    variables: {
-      name,
-      file: new Blob(),
-      width: imageWidth,
-      height: imageHeight,
-      datasetId: "mocked-dataset-id",
-    },
-  });
-
-  const {
-    data: {
-      createImage: { id },
-    },
-  } = mutationResult;
-
-  return id;
-};
-
 test("File should be downloaded when user clicks on Export to COCO and Export", async () => {
-  await createDataset();
-  render(<ExportModal isOpen onClose={() => {}} />, { wrapper });
+  window.URL.createObjectURL = jest.fn();
```

Comment:
What is this for exactly?
Also, if it's really needed, can you also make sure that the function is being called as expected?

Reply:
It's a mock for [this](https://github.com/labelflow/labelflow/blob/74253549bb57f8142177b8d8b034e417bc7d2f46/typescript/web/src/components/export-button/export-modal/export-dataset.ts#L61)